misc/package/Data Files/MWSE/core/lib/DependencyManager/v1/init.lua

```diff
@@ -41,209 +40,282 @@ function DependencyManager.new(e)
+            self.logger:debug("Could not find version in dependency metadata file: %s", metadataPath)
+            return false, string.format("Could not find version in dependency metadata file: %s", metadataPath)
         end
-        return self:doVersionCheck(dependency, metadata.version), metadata.version
+        return self:doVersionCheck(dependency, metadata.version)
     end
 end
 
-function DependencyManager:doVersionFileCheck(dependency)
-    local versionFile = self:getFile(dependency.versionFile)
+function DependencyManager:doVersionFileCheck(dependency, versionFilePath)
+    local versionFile = io.open(versionFilePath, "r")
     if not versionFile then
-        self.logger:error("Could not find dependency version file: %s", dependency.versionFile)
-        return false
+        self.logger:error("Could not find dependency version file: %s", versionFilePath)
+        return false, string.format("Could not find dependency version file: %s", versionFilePath)
     else
         local version = ""
         for line in versionFile:lines() do -- Loops over all the lines in an open text file
             version = line
         end
         if version == "" then
-            self.logger:error("Could not find version in dependency version file: %s", dependency.versionFile)
-            return true
+            self.logger:error("Could not find version in dependency version file: %s", versionFilePath)
+            return false, string.format("Could not find version in dependency version file: %s", versionFilePath)
         end
-        return self:doVersionCheck(dependency, version), version
+        return self:doVersionCheck(dependency, version)
     end
 end
 
+---@param dependency DependencyManager.Dependency
+function DependencyManager:doLuaModCheck(dependency)
+
+    local path = dependency.luaMod:gsub("[/.]", "\\"):lower()
+    local packagePaths = package.path:gsub("%?%.lua", "?")
+
+    for packagePath in packagePaths:gmatch("[^;]+") do
+        local fullPath = packagePath:gsub("?", path .. "\\")
+        self.logger:debug("checking path: %s", fullPath)
+
+        -- if no version is specific, just check the folder exists
+        if dependency.version == nil then
+            self.logger:debug("No target version, check if '%s' folder exists", fullPath)
+            local folderExists = lfs.directoryexists( tes3.installDirectory .. fullPath)
+            if folderExists then
+                return true
+            else
+                self.logger:debug("Folder does not exist: %s", fullPath)
+                return false, string.format("Could not find dependency folder: %s", fullPath)
+            end
+        end
+
+        --Look for metadata
+        local metadataPath = fullPath .. "metadata.json"
+        -- .\\Data Files\\MWSE\\core\\mods\\Ashfall\\
+        local metadataExists = lfs.fileexists( tes3.installDirectory .. metadataPath)
+        -- C:\\Game\\Morrowind\\Data Files\\MWSE\\core\\mods\\Ashfall\\metadata.json
```

Comment:
Are just the comments wrong? Since mods should be inside `Data Files\\MWSE\\mods\\` folder.

Reply:
Oh, I see these are split from package.path. Then this is all right.